samza-kv/src/main/scala/org/apache/samza/storage/kv/BaseKeyValueStorageEngineFactory.scala

```diff
@@ -162,4 +184,28 @@ trait BaseKeyValueStorageEngineFactory[K, V] extends StorageEngineFactory[K, V]
       keyValueStorageEngineMetrics, batchSize, () => clock.nanoTime())
   }
 
+  def createCachedStore[K, V](enableCache: Boolean, storeName: String, registry: MetricsRegistry,
+    underlyingStore: KeyValueStore[K, V], cacheSize: Int, batchSize: Int): KeyValueStore[K, V] = {
+    // maybe wrap with caching
+    val maybeCachedStore = if (enableCache) {
+      val cachedStoreMetrics = new CachedStoreMetrics(storeName, registry)
+      new CachedStore(underlyingStore, cacheSize, batchSize, cachedStoreMetrics)
+    } else {
+      underlyingStore
+    }
+    maybeCachedStore
+  }
+
+  def createLargeMessageSafeStore(underlyingStore: KeyValueStore[Array[Byte], Array[Byte]], storeName: String,
```

Comment:
Why does this need to be a separate method? It's should be the same amount of code to inline this.

Reply:
I think I had suggested this. It wasn't for saving code. It was just to help with readability. When everything was inlined, I thought it was a bit difficult to keep track of all of the different "store" variables through all of the wiring being done. I thought helped to isolate some things so the reader doesn't have to keep track of everything all at once.

Reply:
Yeah, agree that the new code is significantly more complex and we should make it more readable. But I think we need to go further than this and fix the store wrapping pattern and make it consistent. This PR introduces a new `toBeXXX` variable naming convention + var reassignments, which hurt readability a lot more. See also the boolean parameter on the other extracted method.

E.g., the call site for this vs the earlier version will be:
```
createLargeMessageSafeStore(underlyingStore, storeName, dropLargeMessage, maxMessageSize)
new LargeMessageSafeStore(underlyingStore, storeName, dropLargeMessage, maxMessageSize)
```
Which is basically the same from a readability perspective.

Reply:
Ah, that's a good point for the `createLargeMessageSafeStore` case. I guess I was thinking of some of the other helpers.

Reply:
Done.